### ISSUE TITLE
[Extension Feature Request] Add Todoist defaults values in preferences

### ISSUE THREAD

**Issue by emileaublet** [raycast/extensions#714]
# Extension – [Todoist](https://github.com/raycast/extensions/tree/main/extensions/todoist).

Author: @thomaslombart @thomaspaulmann 

Add `default values` in Todoist preferences (e.g. default priority, project, due date etc.) to make task creation faster.  

**Describe the feature and the current behavior/state.**

Right now it defaults to the first item in the option list. If you create one and then a second one, the UI will keep the previous values **but that will not be saved** .

**Comment by thomaslombart** [comment]
It's not possible to add certain values in the preferences of the Todoist's extension because they can be dynamic, like the projects or the labels. While it's still possible to do it for the due dates and priorities, I would find it to be confusing to have my creation task form with certain values filled and not the others. What do you think?

**Comment by thomaspaulmann** [comment]
@thomaslombart not sure if you already use the `storeValue` prop on the `Form` components. It allows you to smartly remember the last selected value, which is handy for some fields like projects. This way you can create a few todos for the same project faster. Maybe worth experimenting with?

**Comment by emileaublet** [comment]
In any case, there's currently a bug where the previous value still **look** selected (e.g. due date) after creating a task, but is actually not selected.

**Comment by thomaslombart** [comment]
That's an interesting one @thomaspaulmann. I'm not sure it should be used for every field, but projects sound like a good use-case.

@emileaublet I think that's one is due to an internal bug on Raycast where it's not possible to properly reset the date. I'm not sure of that, so I'll investigate it this week.

**Comment by thomaslombart** [comment]
I did some investigations:
- I can't really use `storeValue` because I'm making use of controlled form elements to properly reset the values afterward. I could go with a mix of uncontrolled and controlled form elements though. I'm wondering if using the [storage API](https://developers.raycast.com/api-reference/storage) makes more sense here.
- It seems the bug for resetting the due date is still a thing. I'm setting it to `undefined` after creating a task but for some reason, the due date isn't reset.

On the use case itself, I checked Todoist's quick-add feature and they don't seem to save any kind of values. I, myself for my workflow wouldn't like to save values because I'm quick-adding any kind of tasks for any kind of project. While the feature request makes sense for preferences (because the user has decided to do so), I don't think it's the same to automatically save it for him given the number of possible workflows. 

The only way to implement this feature would be to be able to sync Todoist's user data in the background with the Raycast extension + be able to dynamically change the user preferences but I don't think that's feasible for the time being.

I'd suggest putting this issue on hold or closing it pending the improvements needed to make it work 🙂

**Comment by thomaspaulmann** [comment]
@thomaslombart how does `storeValue` behave with controlled form elements? They should still work and set the value properly. You just need to make sure to not pass in `undefined`. If that doesn't work, we should take a look. It would be a shame when `storeValue` doesn't work with your setup.

**Comment by thomaslombart** [comment]
@thomaspaulmann It seems that `storeValue` works properly when using a `SubmitFormAction` which is not the case in the "Create Task" command. The reason why `SubmitFormAction` is not used here is that there are two possible actions for the form: 
- Create a task
- Create a task and go to project

I guess it's possible to do so by replacing the "Create a task and go to project" action with a [Toast with action](https://developers.raycast.com/api-reference/user-interface/toast#example-3) though I didn't look at it yet.

**Comment by thomaspaulmann** [comment]
> 

You should be able to use the `SubmitFormAction` regardless. It has an `onSubmit` callback that gives you the form values but you can ignore it. It's just a thin wrapper to get the values. But this would make sure that we store the submitted value. Though, is this clearly communicated in our documentation?

**Comment by thomaslombart** [comment]
The problem is that it seems we can use `SubmitFormAction` only once in the `ActionPanel`. However, the "Create Task" command has two action items which is a problem since only one is rendered under the hood. I took two screenshots to make this point clearer:

<img width="789" alt="CleanShot 2022-02-03 at 17 14 20@2x" src="https://user-images.githubusercontent.com/16003285/152382280-19c3f896-4625-4758-847f-91061e956c8c.png">

<img width="770" alt="CleanShot 2022-02-03 at 17 14 39@2x" src="https://user-images.githubusercontent.com/16003285/152382274-f93777a5-89ea-468c-918d-5816c6e3d32d.png">

**Comment by thomaspaulmann** [comment]
Oh interesting... This smells like a bug! Could you create a GH issue for it?

**Comment by thomaslombart** [comment]
Oh, I thought it was designed that way and it wasn't actually a bug! Thanks for creating the GitHub issue.

**Comment by thomaslombart** [comment]
@emileaublet I worked a bit on the extension this weekend. I added various improvements including the one you requested. Now creating a task saves the project and the labels for future usage. I didn't include the priority and the due date because I think they can vary a lot depending on the task. 

I did the same when creating a project with the color.
